packages/pigment-css-react/HOW_PIGMENT_CSS_WORKS.md

```diff
@@ -0,0 +1,152 @@
+    text: {
+      secondary: '#00e676',
+    },
+  },
+  typography: {
+    body1: {
+      fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
+      fontWeight: 400,
+      fontSize: 16,
+      lineHeight: 1.5,
+      // ...others
+    },
+  },
+};
+
+const pigmentConfig = {
+  theme: customTheme,
+};
+```
+
+When the source file above (`app.js`) goes through the bundler transform, the bundler plugin of Pigment CSS looks for the call site of `css` imported from `@pigment-css/react` and prepares an intermediate source file to get the actual values of the arguments of the `css` call. In this case, that intermediate code might look like:
+
+```js
+const _exp1 = ({ theme }) => ({
+  lineHeight: '1.4375em',
+  padding: 0,
+  position: 'relative',
+  color: (theme.vars || theme).palette.text.secondary,
+  ...theme.typography.body1,
+});
+
+module.exports = {
+  _exp1,
+};
+```
+
+The above code is then evaluated through the use of `node:module` [module](https://nodejs.org/docs/v20.11.1/api/modules.html#module) which returns the evaluated values of the parameters.
+Now that it has access to the actual values of the styles, Pigment CSS does source code transformation in-place to remove the `css` call from the source and replaces it with a static class name string:
+
+```js
+// app.js
+import { css } from '@pigment-css/react';
+
+const testClass = 'c1aiqtje';
+```
+
+At the same time, it generates the CSS string for the above styles using `emotion` at build time and through the bundler specific plugin, makes it part of the output CSS:
+
+```js
+// app.js
+import { css } from '@pigment-css/react';
+import './app.pigment.css';
+
+const testClass = 'c1aiqtje';
+```
+
+Above, you can assume the import of `./app.pigment.css` to be a virtual module (this file doesn't actually exist on the file system) with the contents generated in-memory. This varies across various bundler implementations but the general idea is the same. Here, the contents of the `app.pigment.css` file would be:
+
+```css
+.c1aiqtje {
+  padding: 0;
+  position: relative;
+  color: #00e676;
+  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
+  font-weight: 400;
+  font-size: 16;
+  line-height: 1.5;
+}
+```
+
+If the same was done using the `styled` call instead of `css` for the source:
+
+```js
+// app.js
+import { styled } from '@pigment-css/react';
+
+const Paragraph = styled.p(({ theme }) => ({
+  lineHeight: '1.4375em',
+  padding: 0,
+  position: 'relative',
+  color: (theme.vars || theme).palette.text.secondary,
+  ...theme.typography.body1,
+}));
+```
+
+The transformed output would look like:
+
+```js
+// app.js
+import { styled } from '@pigment-css/react';
+
+const Paragraph = styled('p')({
+  baseClasses: ['c1aiqtje'],
+});
+```
+
+You can notice above that now the runtime code, that finally ends up in the output bundle, doesn't involve any kind of CSS generation and it only needs to deal with primitive strings to apply it to the `p` that it'd render in the end.
```

Comment:
I feel like this can be replaced with an ending for the `css` explanation. Maybe creating an example section for other processors would be better.

Like:
```
## Example transformation

### styled

…

### keyframes

### generateAtomics
```